File: web_driver.py
import logging
from selenium import webdriver
from selenium.webdriver.chrome.options import Options

class CustomWebDriver:
    def __init__(self):
        self._driver = None
        self._implicit_wait_time = 20

    def __enter__(self):
        logging.info("Open browser")
        options = Options()
        options.add_argument('--disable-blink-features=AutomationControlled')
        self._driver = webdriver.Chrome(options=options)
        self._driver.implicitly_wait(self._implicit_wait_time)
        self._driver.execute_script("Object.defineProperty(navigator, 'webdriver', {get: () => undefined})")
        self._driver.execute_cdp_cmd('Network.setUserAgentOverride', {"userAgent": 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.53 Safari/537.36'})
        return self._driver

    def __exit__(self, exc_type, exc_value, exc_tb):
        logging.info("Close browser")
        if self._driver:
            self._driver.quit()

File: TerminGenie/TerminGenie/Webdriver.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using System;

namespace TerminGenie
{
    public class CustomWebDriver : IDisposable
    {
        private IWebDriver _driver;
        private readonly TimeSpan _implicitWaitTime = TimeSpan.FromSeconds(20);

        public CustomWebDriver()
        {
            InitializeDriver();
        }

        private void InitializeDriver()
        {
            var options = new ChromeOptions();
            // Mimic browser behavior to avoid detection
            options.AddArgument("--disable-blink-features=AutomationControlled");
            options.AddExcludedArgument("enable-automation");
            options.AddAdditionalOption("useAutomationExtension", false);
            options.AddUserProfilePreference("credentials_enable_service", false);
            options.AddUserProfilePreference("profile.password_manager_enabled", false);

            // Optionally, specify the ChromeDriver directory if it's not in your PATH
            // var driverService = ChromeDriverService.CreateDefaultService("path/to/your/chromedriver");
            // _driver = new ChromeDriver(driverService, options);

            _driver = new ChromeDriver(options);
            _driver.Manage().Timeouts().ImplicitWait = _implicitWaitTime;

            // Use JavaScript to set the navigator.webdriver flag to undefined
            ((IJavaScriptExecutor)_driver).ExecuteScript("Object.defineProperty(navigator, 'webdriver', {get: () => undefined})");
        }

        public IWebDriver Driver => _driver;

        public void Dispose()
        {
            if (_driver != null)
            {
                _driver.Quit();
                _driver.Dispose();
            }
        }
    }
}
